internal/adapters/objectset.go

```diff
@@ -0,0 +1,325 @@
+package adapters
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1alpha1 "package-operator.run/apis/core/v1alpha1"
+)
+
+const (
+	pausedByParentAnnotation = "package-operator.run/paused-by-parent"
+	pausedByParentTrue       = "true"
+)
+
+type ObjectSetAccessor interface {
+	ClientObject() client.Object
+	GetTemplateSpec() corev1alpha1.ObjectSetTemplateSpec
+	SetTemplateSpec(templateSpec corev1alpha1.ObjectSetTemplateSpec)
+	GetPrevious() []corev1alpha1.PreviousRevisionReference
+	SetPreviousRevisions(prev []ObjectSetAccessor)
+	GetPhases() []corev1alpha1.ObjectSetTemplatePhase
+	SetPhases(phases []corev1alpha1.ObjectSetTemplatePhase)
+	GetConditions() *[]metav1.Condition
```

Comment:
What's the purpose of returning a slice pointer here?

Reply:
Initially, there were two variants of this method for the objectdeployment and objectset controllers. The objectSet controller package uses some functions that require this to return a pointer, e.g.: [objectset_controller.go#L342](https://github.com/package-operator/package-operator/blob/fc980923be8b4b4c3f9b35b1c285bbfb66387aef/internal/controllers/objectsets/objectset_controller.go#L342). So I just kept the one variant.

Reply:
Makes sense!